backend/src/main/scala/cromwell/backend/BackendWorkflowInitializationActor.scala

```diff
@@ -118,34 +117,34 @@ trait BackendWorkflowInitializationActor extends BackendWorkflowLifecycleActor w
     * - It would be nice to memoize as much of the work that gets done here as possible so it doesn't have to all be
     *   repeated when the various `FooRuntimeAttributes` classes are created, in the spirit of #1076.
     */
-  private def validateRuntimeAttributes: Future[Unit] = {
-
-    coerceDefaultRuntimeAttributes(workflowDescriptor.workflowOptions) match {
-      case Success(_) =>
-
-//        def defaultRuntimeAttribute(name: String): Option[WomValue] = {
-//          defaultRuntimeAttributes.get(name)
-//        }
-
-        def badRuntimeAttrsForTask(task: TaskDefinition) = {
-          // TODO WOM: https://github.com/broadinstitute/cromwell/issues/2606
-//          runtimeAttributeValidators map { case (attributeName, validator) =>
-//            val value = task.runtimeAttributes.attributes.get(attributeName) orElse defaultRuntimeAttribute(attributeName)
-//            attributeName -> ((value, validator(value)))
-//          } collect {
-//            case (name, (value, false)) => RuntimeAttributeValidationFailure(task.name, name, value)
-//          }
-          Seq.empty
+  def validateRuntimeAttributes: Try[Unit] = {
+
+    def checkForBadRuntimeAttributes(defaultRuntimeAttributes: Map[String, WomValue]): Try[Unit] = {
+      def badRuntimeAttrsForTask(task: TaskDefinition) = {
+        runtimeAttributeValidators map {
+          case (attributeName, validator) =>
+            val value: Option[WomExpression] =
+              task.runtimeAttributes.attributes.get(attributeName) orElse
+                defaultRuntimeAttributes.get(attributeName).map(_.asWomExpression)
+            attributeName -> ((value, validator(value)))
+        } collect {
+          case (name, (value, false)) => RuntimeAttributeValidationFailure(task.name, name, value)
         }
+      }
 
-        calls map { _.callable } flatMap badRuntimeAttrsForTask match {
-          case errors if errors.isEmpty => Future.successful(())
-          case errors => Future.failed(RuntimeAttributeValidationFailures(errors.toList))
-        }
-      case Failure(t) => Future.failed(t)
+      calls map { _.callable } flatMap badRuntimeAttrsForTask match {
+        case errors if errors.isEmpty => Try(())
```

Comment:
I just have a feeling that will be a `Success`

Reply:
Says Mr `Option(everything)`?

Reply:
I like that title, it makes it sound like I won a beauty pageant.